src/compilers/compiler_common.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "compiler.hpp"
+#include "platform.hpp"
+#include <dxc/dxcapi.h>
+
+void Logger::Log(const char *msg) const
+{
+    std::lock_guard lock(m_lock);
+    m_buildLog += msg;
+}
+
+static ProgramBinary::Kernel const& FindKernelInfo(std::vector<ProgramBinary::Kernel> const& kernels, const char *name)
+{
+    auto iter = std::find_if(kernels.begin(), kernels.end(), [name](ProgramBinary::Kernel const& k) { return strcmp(k.name, name) == 0; });
+    return *iter;

```

Comment:
Can this search ever fail?

Reply:
No. The only way to get a `CompiledDxil` object is by taking a SPIR-V `ProgramBinary` and running it through the `spirv_to_dxil` process in the compiler DLL, which also uses the name to do a search through the raw metadata (as opposed to the abstrcted metadata that's used here). I can comment and add an assertion.